flashinfer/moe_ep/kernel_src/cutedsl_megamoe/src/moe_bf16_glu/mega_reference_bf16.py

```
@@ -0,0 +1,1575 @@
+        """
+        valid_ab_dtypes = {
+            cutlass.Float16,
+            cutlass.BFloat16,
+            cutlass.TFloat32,
+            cutlass.Uint8,
+            cutlass.Int8,
+            cutlass.Float8E4M3FN,
+            cutlass.Float8E5M2,
+        }
+        if a_dtype not in valid_ab_dtypes or b_dtype not in valid_ab_dtypes:
+            raise testing.CantImplementError(
```

Ruff (0.15.21)

[error] Undefined name `testing`

(F821)